File: Baekjoon/Greedy/1339.py
# https://www.acmicpc.net/problem/1339

n = int(input())
word_list = [input() for _ in range(n)]
word_dict = {}

for word in word_list:
    power = len(word) - 1
    for i in word:
        if i in word_dict:
            word_dict[i] += pow(10, power)
        else:
            word_dict[i] = pow(10, power)
        power -= 1

word_dict = sorted(word_dict.values(), reverse = True)
answer = 0
num = 9

for word in word_dict:
    answer += word * num
    num -= 1

print(answer)

File: Baekjoon/Greedy/1339.cpp
// https://www.acmicpc.net/problem/1339

#include <iostream>
#include <vector>
#include <algorithm>
#include <string>
using namespace std;

int main() {
    int N;
    cin >> N;
    vector<string> words(N);
    for (int i = 0; i < N; i++) {
        cin >> words[i];
    }

    vector<int> alpha(26);
    for (int i = 0; i < N; i++) {
        int k = 1;
        for (int j = words[i].size() - 1; j >= 0; j--) {
            alpha[words[i][j] - 'A'] += k;
            k *= 10;
        }
    }

    sort(alpha.begin(), alpha.end(), greater<int>());

    int ans = 0;
    int num = 9;
    for (int i = 0; i < 26; i++) {
        if (alpha[i] == 0) {
            break;
        }
        ans += alpha[i] * num;
        num--;
    }
    cout << ans << '\n';

    return 0;
}